src/DataTypes/DataTypesDecimal.cpp

```diff
@@ -58,7 +58,7 @@ void DataTypeDecimal<T>::serializeText(const IColumn & column, size_t row_num, W
 }
 
 template <typename T>
-bool DataTypeDecimal<T>::tryReadText(T & x, ReadBuffer & istr, UInt32 precision, UInt32 scale)
+bool NO_SANITIZE_UNDEFINED DataTypeDecimal<T>::tryReadText(T & x, ReadBuffer & istr, UInt32 precision, UInt32 scale)
```

Comment:
Maybe it doesn't work for template classes?

Reply:
Maybe.

Also there is another way - to enable overflow check (and throw exception on overflow).
There are the following considerations:
- it will be cheap but not free;
- we need performance test for formats but it is disabled at this moment of time;
- maybe I can prepare another performance test for decimal parse;
- overflow check is better from user perspective but worth to note that it will have some false negatives - if the number exceeds maximum number of digits but still don't overflow inside integer data type.